scripts_python/create_hdf5_dataset.py

```diff
@@ -0,0 +1,255 @@
+
+
+def _initialize_hdf5_database(database_path, force, name):
+    # Create hdf5 dir or clean existing one
+    hdf5_dir = Path(database_path, "hdf5")
+    if hdf5_dir.is_dir():
+        if force:
+            logging.info("Careful! Deleting existing hdf5 data folder: "
+                         "{}".format(hdf5_dir))
+            shutil.rmtree(str(hdf5_dir))
+        else:
+            raise FileExistsError("hdf5 data folder already exists: {}. Use "
+                                  "force to allow overwrite.".format(hdf5_dir))
+    hdf5_dir.mkdir()
+
+    # Define dabase name
+    if name:
+        hdf5_filename = name
+    else:
+        hdf5_filename = "{}".format(
+            datetime.datetime.now().strftime("%Y_%m_%d_%H%M%S"))
+
+    return hdf5_dir, hdf5_filename
+
+
+def _add_all_subjs_to_database(args, chosen_subjs: List[str],
+                               groups_config: Dict, hdf5_dir, hdf5_filename,
+                               dwi_ml_dir):
+    """
+    Generate a dataset from a group of dMRI subjects with multiple bundles.
+    All bundles are merged as a single whole-brain dataset in voxel space.
+    If wished, all intermediate steps are saved on disk in the hdf5 folder.
+    """
+    # Cast space from str to Space
+    if args.space == 'Space.Vox':
+        space = Space.Vox
+    elif args.space == 'Space.RASMM':
+        space = Space.RASMM
+    else:
+        space = Space.VOXMM
+
+    # Add data to database
+    hdf5_file_path = hdf5_dir.joinpath("{}.hdf5".format(hdf5_filename))
+    with h5py.File(hdf5_file_path, 'w') as hdf_file:
+        # Save version and configuration
+        hdf_file.attrs['version'] = 1
+        hdf_file.attrs['chosen_subjs'] = chosen_subjs
+        hdf_file.attrs['groups'] = str(groups_config)
+        if args.step_size is not None:
+            hdf_file.attrs['step_size'] = args.step_size
+        else:
+            hdf_file.attrs['step_size'] = 'Not defined by user'
+        hdf_file.attrs['bundles'] = [b for b in args.bundles]
+        hdf_file.attrs['space'] = args.space
+
+        # Add data one subject at the time
+        logging.info("Processing {} subjects : "
+                     "{}".format(len(chosen_subjs), chosen_subjs))
+        for subj_id in chosen_subjs:
+            logging.info("\n *Processing subject: {}".format(subj_id))
+            subj_input_dir = dwi_ml_dir.joinpath(subj_id)
+
+            # Add subject to hdf database
+            subj_hdf = hdf_file.create_group(subj_id)
+
+            # Prepare subject folder for intermediate files
+            subj_intermediate_path = hdf5_dir.joinpath(subj_id)
+            if args.save_intermediate:
+                subj_intermediate_path.mkdir()
+
+            # Find subject's mask
+            if args.mask:
+                subj_mask_file = subj_input_dir.joinpath(args.mask)
+                subj_mask_data = nib.load(subj_mask_file)
+                subj_mask_data = subj_mask_data.get_fdata().astype(np.bool)
```

Comment:
For a mask you should use `np.asanyarray(img.dataobj).astype(np.bool)`. This way you don't cast everything to float64 by default when calling `get_fdata()`